overlord/state/task.go

```diff
@@ -196,26 +206,59 @@ func (t *Task) Status() Status {
 	return t.status
 }
 
+func (t *Task) changeStatus(old, new Status) {
+	t.status = new
+	if !old.Ready() && new.Ready() {
+		t.readyTime = timeNow()
+	}
+	chg := t.Change()
+	if chg != nil {
+		chg.taskStatusChanged(t, old, new)
+	}
+}
+
 // SetStatus sets the task status, overriding the default behavior (see Status method).
 func (t *Task) SetStatus(new Status) {
+	if new == WaitStatus {
+		// Disallow going into WaitStatus, the Wait() API should be used instead.
+		return
```

Comment:
we should probably panic here instead of just returning, under the assumption that any code doing SetStatus must have unit tests

Reply:
Ah good point, added this!